### PR TITLE
also handle ValueError nicely

### PR DESCRIPTION
**Pull request by dirk-thomas**
Follow up of #24.

This handles `ValueErrors` as well. To reproduce try:

```
ros2 topic pub initialpose geometry_msgs/PoseWithCovarianceStamped "header:
  stamp:
    sec: 'sec'
    nanosec: 0.5
  frame_id: 'map'   
pose:
  pose:
    position: {x: 5.5, y: 4.5, z: 0.0}
    orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}
  covariance: [0.5, 0.5, 0.0, 0.0, 0.0, 0.0, 0.0, 0.5, 0.5, 0.0, 0.0, 0.0, 0.0, 0.0,    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
    0.0, 0.0, 0.0, 0.0, 0.0, 0.0]"
```

### CODE REVIEW

File: ros2topic/ros2topic/verb/pub.py [hunk 1]
```diff
@@ -99,6 +99,9 @@ def set_msg_fields(msg, values):
                 raise SetFieldError(
                     '{field_name}.{e.field_name}'.format_map(locals()),
                     e.exception)
+        except ValueError as e:
+            raise SetFieldError(
+                '{field_name}'.format_map(locals()), e)
```
Comment (wjwwood) [inline ros2topic/ros2topic/verb/pub.py]: Do you not have to do `**locals()`?

Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/pub.py]: No, not with `format_map`. There is already plenty of other code using this snippet.

Comment (wjwwood) [inline ros2topic/ros2topic/verb/pub.py]: Ah sorry, thought it was just format, lgtm.

Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/pub.py]: Looking at the line again the whole use of `format_map` is pretty unnecessary. I used it for the field name concatenation in another line but here it only contains `field_name`. I will update it to use the literal variable instead...

**Pull request merged**